coffee-module/coffee-module-redisstream/src/main/java/hu/icellmobilsoft/coffee/module/redisstream/service/RedisStreamService.java

```diff
@@ -0,0 +1,378 @@
+
+    private String group;
+
+    /**
+     * Stream key, calculated by {@link #group}
+     * 
+     * @return Sream key
+     */
+    public String streamKey() {
+        return getGroup() + "Stream";
+    }
+
+    /**
+     * Publish one element to stream with values. Stream max size is trimmed by config. This is equivalent to redis console:
+     * 
+     * <pre>
+     * XADD streamKey * key1 value1 key2 value2...
+     * </pre>
+     * 
+     * @param values
+     *            Values in stream element
+     * @return Generated ID
+     * @throws BaseException
+     *             Exception
+     */
+    public StreamEntryID publish(Map<String, String> values) throws BaseException {
+        if (values == null) {
+            throw new BaseException(CoffeeFaultType.INVALID_INPUT, "publish values is null");
+        }
+        StreamEntryID streamEntryID = getJedis().xadd(streamKey(), null, values, config.getStreamMaxLen(), false);
+        if (log.isTraceEnabled()) {
+            log.trace("Published streamEntryID: [{0}] into [{1}]", streamEntryID, streamKey());
+        }
+        return streamEntryID;
+    }
+
+    /**
+     * Count elements in stream. This is equivalent to redis console:
+     * 
+     * <pre>
+     * XLEN streamKey
+     * </pre>
+     * 
+     * @return elements count
+     */
+    public Long count() {
+        Long count = getJedis().xlen(streamKey());
+        if (log.isTraceEnabled()) {
+            log.trace("[{0}] stream have [{1}] elements", streamKey(), count);
+        }
+        return count;
+    }
+
+    /**
+     * Check group exist. This is equivalent to redis console:
+     * 
+     * <pre>
+     * XINFO GROUPS streamKey
+     * </pre>
+     * 
+     * and search checking group
+     * 
+     * @return true if exist
+     */
+    public boolean existGroup() {
+        try {
+            List<StreamGroupInfo> info = getJedis().xinfoGroup(streamKey());
+            return info.stream().map(StreamGroupInfo::getName).anyMatch(name -> StringUtils.equals(getGroup(), name));
+        } catch (JedisDataException e) {
+            // ha nincs kulcs akkor a kovetkezo hiba jon:
+            // redis.clients.jedis.exceptions.JedisDataException: ERR no such key
+            log.info("Redis exception duringchecking group [{0}]: [{1}]", streamKey(), e.getLocalizedMessage());
+            return false;
+        }
+    }
+
+    /**
+     * Checking for existing group. If not exist, then create it, with also creating non-existing stream too. This is equivalent to redis console:
+     * 
+     * <pre>
+     * XINFO GROUPS streamKey
+     * # if not exist group then
+     * XGROUP CREATE streamKey group 0 MKSTREAM
+     * </pre>
+     */
+    public void handleGroup() {
+        if (existGroup() && log.isTraceEnabled()) {
+            log.trace("Group [{0}] already exist", getGroup());
+        } else {
+            String createGroupResult = getJedis().xgroupCreate(streamKey(), getGroup(), new StreamEntryID(), true);
```

Comment:
ide akkor is be fog futni, ha existGroup() true, de a log.isTraceEnabled() false

Reply:
jogos, koszi